### PR TITLE
fix: Groups modal does not include inactive users [WEB-1256]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

In RBAC mode, the admin can add and remove users from groups.
The ticket is to keep deactivated users from being newly added to groups.
Current code uses the `disabled` attribute; this means a deactivated user cannot be added or removed from groups.

I tried filtering out deactivated users from the component, but then the antd.Select component shows an inactive user as their number id ("44").

## Test Plan

As admin of an EE / RBAC server, on `/det/admin/group-management`

Click "New Group" and click under the Users label. You should see a list of many users which you can add to the group
For an existing group with users, click the row's three-dot menu and Edit to see users who can be added or removed from the group

On the users management page, create a test user and deactivate that account.
In the "New Group" modal, that user is greyed out / unselectable.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

WEB-1256

### CODE REVIEW

File: webui/react/src/components/CreateGroupModal.tsx [hunk 1]
```diff
@@ -193,7 +193,7 @@ const CreateGroupModalComponent: React.FC<Props> = ({ onClose, users, group }: P
           <Form.Item label={USERS_LABEL} name={USERS_NAME}>
             <Select mode="multiple" optionFilterProp="children" placeholder="Add Users" showSearch>
               {users?.map((u) => (
-                <Select.Option key={u.id} value={u.id}>
+                <Select.Option disabled={!u.isActive} key={u.id} value={u.id}>
```
Comment (hkang1) [inline webui/react/src/components/CreateGroupModal.tsx:196]: **question:** if you created a something like:
```
const activeUsers = (users ?? []).filter((user) => user.isActive);
```
then use the JSX:
```
{activeUsers.map((u) => { ... }}
```
Would you still run into the rendering issue you referred to? If the entry doesn't even exist for `activeUsers` then it should even show up as an antd.Select.Option...

**Review by hkang1 — COMMENTED** [review]
Wondering what specifically happens around the filtering solution. I would prefer it to not even show the deactivate users if possible.

File: webui/react/src/components/CreateGroupModal.tsx (hunk 1 above)
Comment (ClaireNeveu) [reply webui/react/src/components/CreateGroupModal.tsx:196]: IIRC the issue that originally arose was when a user was in a group and later deactivated.

**Review by hkang1 — APPROVED** [review]
Looking at the proposed behavior:

> When managing group memberships, the Users multiselector in the Edit group view does not allow you to add deactivated users but it allows you to remove them

Making the user show up but disabled does fulfill the requirement IMO.

File: webui/react/src/components/CreateGroupModal.tsx [hunk 2]
```diff
@@ -192,11 +194,13 @@ const CreateGroupModalComponent: React.FC<Props> = ({ onClose, users, group }: P
           </Form.Item>
           <Form.Item label={USERS_LABEL} name={USERS_NAME}>
             <Select mode="multiple" optionFilterProp="children" placeholder="Add Users" showSearch>
-              {users?.map((u) => (
-                <Select.Option key={u.id} value={u.id}>
-                  {getDisplayName(u)}
-                </Select.Option>
-              ))}
+              {users
+                ?.filter((u) => u.isActive || currentGroupMembers?.includes(u.id))
```
Comment (hkang1) [inline webui/react/src/components/CreateGroupModal.tsx:198]: **nit suggestion:** our modals seem to re-render a lot, where having this filters user list cached (via `useMemo`) could help the performance slightly.

**Review by hkang1 — APPROVED** [review]
LGTM, good idea to filter the users based on what's already in the form along with active users!

File: webui/react/src/components/CreateGroupModal.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/components/CreateGroupModal.tsx:198]: I tried memo-izing `currentGroupMembers` based on `[form]` and it didn't work, so for now I'll keep it as-is

**Pull request merged**